[MRG] Fix typos in datasets documentation

christopherchoe opened the pull request:
Changed spelling and grammar typos on lines 53 (spelling miscellaneous), 174 (Beginning of sentence not capitalized), 264 (grammar implies images not image).

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.


#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

christopherchoe committed:
Minor Typos in datasets documentation

Changed spelling and grammar typos on lines 53 (spelling miscellaneous), 174 (Beginning of sentence not capitalized), 264 (grammar implies images not image).

qinhanmin2014 commented on doc/datasets/index.rst:
```diff
@@ -171,7 +171,7 @@ near-equal-size classes separated by concentric hyperspheres.
 datasets that are challenging to certain algorithms (e.g. centroid-based
 clustering or linear classification), including optional Gaussian noise.
-They are useful for visualisation. produces Gaussian
+They are useful for visualisation. Produces Gaussian
```
this seems strange,  maybe ``..., and to produce...``?

christopherchoe replied:
I agree, but wasn't sure on what the message was trying to convey so I went with the smallest change possible. And to produce would seem to convey the same message and be clearer.

NicolasHug replied:
Only `make_circles` produces a spherical decision function so the sentence does not apply to both.

How about:

> :func:`make_circles` produces Gaussian data with a spherical decision boundary for binary classification, while :func:`make_moons` produces two interleaving half circles.

christopherchoe replied:
Yeah definitely thought maybe that there might be a different meaning because the sentence seemed to be unfinished. I changed it and changed the formatting by adding backquotes.

> :func:`make_circles ... :func:`make_moons` ...

This was to match with other references in the documentation, please tell me if that shouldn't be the case in this instance. Thank you

christopherchoe committed:
Clarification on visualization

Changed to accurately represent the visualization of :func:make_circles and :func:make_moons

NicolasHug commented on doc/datasets/index.rst:
```diff
@@ -171,8 +171,7 @@ near-equal-size classes separated by concentric hyperspheres.
 clustering or linear classification), including optional Gaussian noise.
-They are useful for visualisation. produces Gaussian
-data with a spherical decision boundary for binary classification.
+They are useful for visualisation. :func:`make_circles` produces Gaussian data with a spherical decision boundary for binary classification, while :func:`make_moons` produces two interleaving half circles.
```
please avoid long lines: split this into multiple lines with approximately 80 characters

christopherchoe replied:
Fixed, thanks for the input!

NicolasHug reviewed (approved):
line length issue, LGTM otherwise!

christopherchoe committed:
Line Length Fix

Fixed documentation to not go over 80 characters

qinhanmin2014 reviewed (approved):
LGTM, thanks @christopherchoe
